ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -136,6 +138,40 @@ public boolean setReplicaSealedState(ReplicaId replicaId, boolean isSealed) {
     }
   }
 
+  @Override
+  public boolean setReplicaStoppedState(List<ReplicaId> replicaIds, boolean isStopped) {
+    Set<String> replicasToUpdate = new HashSet<>();
+    for (ReplicaId replicaId : replicaIds) {
+      if (!(replicaId instanceof AmbryReplica)) {
+        throw new IllegalArgumentException(
+            "HelixParticipant only works with the AmbryReplica implementation of ReplicaId");
+      }
+      replicasToUpdate.add(replicaId.getPartitionId().toPathString());
```

Comment:
should this be called `partitionsToUpdate`?

Reply:
actually, i think `replicasToUpdate` might be slightly better because the `stopped` list is used for single datanode not for cluster-wide use. Hence, we are more interested in the replicas residing on this datanode and don't care about other replicas of this specific partition. Does it sound good for you?